--- branches/KDE/3.5/kdeadmin/knetworkconf/knetworkconf/kcm_knetworkconfmodule.desktop	2008/06/25 16:51:13	824342
+++ branches/KDE/3.5/kdeadmin/knetworkconf/knetworkconf/kcm_knetworkconfmodule.desktop	2008/11/14 08:42:18	884099
@@ -8,7 +8,7 @@
 Type=Application
 Exec=kcmshell kcm_knetworkconfmodule
 Icon=knetworkconf
-DocPath=knetworkconf/index.docbook
+DocPath=knetworkconf/index.html
 
 Comment=Configure Network Settings
 Comment[ar]=إعداد تعيينات الشبكة

--- branches/KDE/3.5/kdeadmin/knetworkconf/knetworkconf/knetworkconf.desktop	2008/06/25 16:51:13	824342
+++ branches/KDE/3.5/kdeadmin/knetworkconf/knetworkconf/knetworkconf.desktop	2008/11/14 08:42:18	884099
@@ -3,7 +3,7 @@
 Type=Application
 Exec=kdesu knetworkconf -caption "%c" %i %m
 Icon=knetworkconf
-DocPath=knetworkconf/index.docbook
+DocPath=knetworkconf/index.html
 Comment=Configure TCP/IP settings
 Comment[ar]=إعداد تعيينات ميفاق التحكم بلنقل/ميفاق الإنترنت
 Comment[be]=Наставіць TCP/IP
