diff -u a/kopete/libkopete/avdevice/videodevice.h b/kopete/libkopete/avdevice/videodevice.h
--- a/kopete/libkopete/avdevice/videodevice.h   2012-04-03 22:50:35.000000000 +0200
+++ b/kopete/libkopete/avdevice/videodevice.h   2012-04-03 22:46:25.000000000 +0200
@@ -52,7 +52,7 @@

 #include <linux/fs.h>
 #include <linux/kernel.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #define VIDEO_MODE_PAL_Nc  3
 #define VIDEO_MODE_PAL_M   4
 #define VIDEO_MODE_PAL_N   5
--
