diff -Naur kdepim-3.5.10.old/indexlib/memvector.tcc kdepim-3.5.10/indexlib/memvector.tcc
--- kdepim-3.5.10.old/indexlib/memvector.tcc	2008-02-13 10:39:38.000000000 +0100
+++ kdepim-3.5.10/indexlib/memvector.tcc	2009-09-22 10:49:26.000000000 +0200
@@ -30,13 +30,13 @@
 void memvector<T>::resize( size_type n_s ) {
 	if ( size() >= n_s ) return;
 
-	data_->resize( n_s * byte_io::byte_lenght<value_type>() + byte_io::byte_lenght<unsigned>() );
+	using namespace byte_io;
+	data_->resize( n_s * byte_lenght<value_type>() + byte_lenght<unsigned>() );
 	iterator p_end = end();
-	byte_io::write<unsigned>( data_->rw_base( 0 ), n_s );
-	while ( operator !=<unsigned, unsigned>(p_end, end()) ) {
+	write<unsigned>( data_->rw_base( 0 ), n_s );
+	while ( p_end != end() ) {
 		*p_end = value_type();
 		++p_end;
-		
 	}
 }
 