From 1692373bba54329dbf520a6067c91d4479d02d04 Mon Sep 17 00:00:00 2001
From: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date: Sat, 06 Aug 2011 08:52:51 +0000
Subject: Fix another FTBFS

---
diff --git a/src/widgets/qsplitter.h b/src/widgets/qsplitter.h
index c159e0f..b5012f6 100644
--- a/src/widgets/qsplitter.h
+++ b/src/widgets/qsplitter.h
@@ -187,11 +187,7 @@ public: //private:
 
     Orientation orient;
     friend class QSplitterHandle;
-
-#ifndef QT_NO_TEXTSTREAM
-    friend Q_EXPORT QTextStream& operator<<( QTextStream&, const QSplitter& );
-    friend Q_EXPORT QTextStream& operator>>( QTextStream&, QSplitter& );
-#endif
+    friend class QTextStream;
 
 private:
 #if defined(Q_DISABLE_COPY)
--
cgit v0.9.0.2-52-g1de3
