--- kdegraphics-3.5.10/kamera/kcontrol/kameradevice.cpp.orig	2006-07-22 08:10:31.000000000 +0000
+++ kdegraphics-3.5.10/kamera/kcontrol/kameradevice.cpp	2015-09-11 20:08:11.000000000 +0000
@@ -118,9 +118,9 @@
 		gp_port_info_list_new(&il);
 		gp_port_info_list_load(il);
 		gp_port_info_list_get_info(il, gp_port_info_list_lookup_path(il, m_path.local8Bit().data()), &info);
-		gp_port_info_list_free(il);
 		gp_camera_set_abilities(m_camera, m_abilities);
 		gp_camera_set_port_info(m_camera, info);
+		gp_port_info_list_free(il);
 
 		// this might take some time (esp. for non-existant camera) - better be done asynchronously
 		result = gp_camera_init(m_camera, glob_context);
@@ -346,7 +346,9 @@
 	}
 	for (int i = 0; i < gphoto_ports; i++) {
 		if (gp_port_info_list_get_info(list, i, &info) >= 0) {
-			if (strncmp(info.path, "serial:", 7) == 0)
-				m_serialPortCombo->insertItem(QString::fromLatin1(info.path).mid(7));
+			char *xpath;
+			gp_port_info_get_path (info, &xpath);
+			if (strncmp(xpath, "serial:", 7) == 0)
+ 				m_serialPortCombo->insertItem(QString::fromLatin1(xpath).mid(7));
 		}
 	}
